data_catalog_backend/services/category_service.py

```diff
@@ -43,3 +52,42 @@ def create_category(self, category: CategoryRequest, user: User) -> Category:
             self.session.rollback()
             raise e
         return category
+
+    def delete_category(self, category_id: uuid.UUID, current_user: User) -> Category:
+        category = self.get_category(category_id)
+
+        if not category:
+            return None
+        if category.resources and len(category.resources) > 0:
+            raise ValueError(
+                "Cannot delete category with resources. Please remove resources first."
+            )
+        self.session.delete(category)
+        try:
+            self.session.commit()
+        except Exception as e:
+            self.session.rollback()
+            raise e
+        print(category)
```

Comment:
We do not use print-statements. This seems to have been forgotten, if you have a need to log something, use a proper logger.